pytorch_lightning/trainer/training_loop.py

```diff
@@ -200,10 +200,10 @@ def on_train_batch_end(self, epoch_output, batch_end_outputs, batch, batch_idx,
         self.trainer.logger_connector.on_train_batch_end()
 
     def reset_train_val_dataloaders(self, model):
-        if self.trainer.train_dataloader is None or not self.trainer.reload_dataloaders_every_epoch:
+        if self.trainer.train_dataloader is None:
             self.trainer.reset_train_dataloader(model)
 
-        if self.trainer.val_dataloaders is None and not self.trainer.reload_dataloaders_every_epoch:
```

Comment:
Should we remove the val reload and leave its responsibility to the val_loop instead ?

Reply:
The val dataloader must be set before training because the train loop depends on it for determining whether to run evaluation